### PR TITLE
fix: Handle missing child dob when generating orders

### PR DESCRIPTION
**Pull request by tomaszpowroznik**

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java [hunk 1]
```diff
@@ -131,6 +131,24 @@ void shouldReturnChildrenDetailsWhenPopulatedChildren() {
...
+            .party(ChildParty.builder()
+                .firstName("John")
+                .lastName("Smith")
+                .build())
+            .build());
+
+        DocmosisChild expectedChild = DocmosisChild.builder()
+            .name("John Smith")
+            .gender("Male")
+            .dateOfBirth(null)
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java:146]: Unless you want to be specific, this shouldn't be needed as null is the default value if you don't assign it a value

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java [hunk 2]
```diff
@@ -131,6 +131,24 @@ void shouldReturnChildrenDetailsWhenPopulatedChildren() {
...
+        List<Element<Child>> listChildren = wrapElements(Child.builder()
+            .party(ChildParty.builder()
+                .firstName("John")
+                .lastName("Smith")
+                .build())
+            .build());
+
+        DocmosisChild expectedChild = DocmosisChild.builder()
+            .name("John Smith")
+            .gender("Male")
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java:145]: Haven't specified a gender in the built child above

Comment (tomaszpowroznik) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java]: stupid me :) that explains test failure ;)done

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java (hunk 1 above)
Comment (tomaszpowroznik) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java:145]: test should not silently assume anything, how do you know if code doesnt set it to "" or "unknown" or "N/A". This test check that it does not

Comment (lewisbirks) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java:145]: Thought so

**Pull request merged**